client/daemon.go

```diff
@@ -242,11 +246,14 @@ func (d daemon) processKey(keyID string) error {
 	if err != nil {
 		return fmt.Errorf("Error marshalling key %s: %s", keyID, err.Error())
 	}
-	err = ioutil.WriteFile(d.keyFilename(keyID), b, defaultFilePermission)
+	err = ioutil.WriteFile(d.tmpFilename(keyID), b, defaultFilePermission)
```

Comment:
From what i remember about this method, it's only called via knox daemon so I think this should be ok. If there is ever a possibility of more than one process writing to this file at the same time, we likely should be using ioutil.TempFile.

Reply:
I had the same thought, but I don't see a downside to using `ioutil.TempFile` right now. One less thing to think about in the future. 😉

Reply:
I had originally used it, then figured it wasn't necessary. adding it back.